src/main/java/com/cloudbees/groovy/cps/TransformerConfiguration.java

```diff
@@ -0,0 +1,27 @@
+package com.cloudbees.groovy.cps;
+
+import com.cloudbees.groovy.cps.impl.CpsClosure;
+import org.codehaus.groovy.ast.ClassNode;
+
+/**
+ * Switches that affect the behaviour of {@link CpsTransformer}
+ *
+ * @author Kohsuke Kawaguchi
+ * @see CpsTransformer#setConfiguration(TransformerConfiguration)
+ */
+public class TransformerConfiguration {
+    private ClassNode closureType = new ClassNode(CpsClosure.class);
+
+    public ClassNode getClosureType() {
+        return closureType;
+    }
+
+    public TransformerConfiguration withClosureType(ClassNode closureType) {
```

Comment:
I think it's useful to define this fully general version as the base contract, because having that version supports all the use cases this library can support without frustrating users that "gee given the way this library is implemented I know it can take `ClassNode` but it just doesn't expose it!"